fix for unreleased buffers under mmap (issue 280)

poke1024 opened the pull request:
This is a proposal for fixing https://github.com/capnproto/pycapnp/issues/280

Using this fix, the demo code reported there does not fail anymore.

This proposal introduces a compatibility breaking change in `from_bytes` by making it a context manager, i.e. changing the way it needs to be called, e.g. through:

```
with record.from_bytes(my_bytes) as data:
     ...
```

I think a context manager approach is best, but I'm not sure about the compatibility breakage regarding `from_bytes`. Alternatively maybe one could add a new variant of `from_bytes`?

haata commented:
Personally, I think this looks like a good change.

Would you mind adding a test case for it? Under pycapnp/test.

poke1024 commented:
I've added a test case and modified the existing `from_bytes` test cases to suit the modified API.

I'm getting the following unrelated test case error:

```
FAILED test/test_load.py::test_bundled_import_hook - ModuleNotFoundError: No module named 'stream_capnp'
```

Not sure how to proceed there.

Note I've isolated the buf release error (without the fix) in this PR: https://github.com/capnproto/pycapnp/pull/282
